### PR TITLE
Process data

### PR DESCRIPTION
**Pull request by ichsansandy**
In these project I Implement :
 
  - User model
    - A method that returns the 3 most recent posts for a given user.
    - A method that updates the posts counter for a user.
  - Post model
    - A method which returns the 5 most recent comments for a given post.
    - A method that updates the comments counter for a post.
    - A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id'
+  has_many :likes
+  has_many :comments
+
+  def recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
+
+  def update_posts_counter
```
Comment (Lucas-Masaba) [inline app/models/user.rb]: - [ ] This method should be in the `post.rb` model as mentioned in the project requirements. Please, move it there
image

[OPTIONAL]
- You can use [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_posts_counter` method private for example:
```ruby
  after_save :update_post_counter

  private

  def update_posts_counter
    author.increment!(:posts_counter)
  end
```

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+
+  def recent_comments(limit = 5)
+    comments.order(created_at: :desc).limit(limit)
+  end
+
+  def update_comments_counter
```
Comment (Lucas-Masaba) [inline app/models/post.rb]: - [ ] This method should be in the `comment.rb` model as mentioned in the project requirements. Please, move it there.

[OPTIONAL]
- Just like I mentioned before, you can use rails active callbacks i.e:
`after_save :update_comments_counter`

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    update(comment_counter: comments.count)
+  end
+
+  def update_likes_counter
```
Comment (Lucas-Masaba) [inline app/models/post.rb]: - [ ] This method should be in the `like.rb` model as mentioned in the project requirements. Please, move it there.

[OPTIONAL]
- Just like I mentioned before, you can use rails active callbacks i.e:
`after_save :update_likes_counter`

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @ichsansandy  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Summarized pull request description :writing_hand: 
- Well written commit messages :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @ichsansandy,

## APPROVED 🟢

## Highlights
✔️ Custom model methods and callbacks

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**